### PR TITLE
feat(kernels): add CUDA layer normalization kernel with CPU references

### PR DESCRIPTION
**Pull request by EffortlessSteven**
## Summary

Add `crates/bitnet-kernels/src/cuda/layer_norm.rs` — a comprehensive CUDA layer normalization kernel module with CPU reference implementations for all variants.

### Types
- **`LayerNormConfig`** — Configuration with eps, elementwise_affine, normalized_shape
- **`LayerNormError`** — Error enum with manual `Display` + `Error` impls (InvalidShape, InvalidEpsilon, GpuUnavailable)

### CPU reference implementations
- **`rms_norm_forward`** — RMS normalization (used by LLaMA/BitNet architectures)
- **`layer_norm_forward`** — Standard LayerNorm with mean subtraction + affine
- **`rms_norm_with_residual`** — Fused RMS norm + residual addition (common in transformer blocks)
- **`group_norm_forward`** — Group normalization (channel-grouped)
- **`batch_norm_forward_inference`** — Batch normalization in eval mode

### CUDA kernels
- Feature-gated `LAYER_NORM_KERNEL_SRC` with five GPU kernels (`layernorm_ext_f32`, `rmsnorm_ext_f32`, `rmsnorm_residual_f32`, `groupnorm_f32`, `batchnorm_inference_f32`)
- GPU launch stubs for all five variants (behind `#[cfg(any(feature = "gpu", feature = "cuda"))]`)

### Tests
- **47 tests total** (42 passing, 5 GPU-gated `#[ignore]`)
- Error display/trait coverage, config validation, numerical correctness, edge cases (empty, large, tiny values), residual fusion equivalence, group-norm ↔ layer-norm consistency

### Registration
- Module registered in `cuda/mod.rs` with `LAYER_NORM_KERNEL_SRC` re-export

## Testing
```
cargo test -p bitnet-kernels --lib --no-default-features --features cpu -- cuda::layer_norm::tests
# 42 passed; 0 failed; 5 ignored
```

Clippy clean (`-D warnings`), formatted with `cargo fmt`.

### CODE REVIEW

**Comment by EffortlessSteven** [comment]
Closing: CUDA scaffold PR  needs GPU hardware validation before merge. These should be re-created with proper testing when CUDA CI lane is active.
